include/Standalone/Transforms.h

```diff
@@ -29,10 +29,23 @@ namespace linalgx {
 FailureOr<SmallVector<Value>> mapToBRGEMMOp(RewriterBase &rewriter,
                                             linalg::LinalgOp linalgOp);
 
-//
-FailureOr<linalg::MatmulOp> mapConvToGemm(RewriterBase &rewriter,
-                                          linalg::LinalgOp linalgOp,
-                                          int64_t rPos, int64_t sPos);
+// Map a convolution to a matmul operation.
+// [N][P][Q][K] += [N][H][W][C] * [R][S][C][K]
+// [N][K’][P][Q][k] += [N][C’][H][W][c] * [K’][C’][R][S][c][k] (blocked)
+class MapConvToMatmul {
+public:
+  MapConvToMatmul() = delete;
+  MapConvToMatmul(int64_t rPos, int64_t sPos) : rPos(rPos), sPos(sPos){};
```

Comment:
IIUC, you're creating this structure amd calling `doIt` straight after on all cases. Why do you need to cache the arguments and not just export a function?

Reply:
Originally was a function: `FailureOr<linalg::MatmulOp>(RewriterBase &rewriter, linalg::LinalgOp linalgOp, int64_t rPos, int64_t sPos)`. But I did not like carrying around `rPos` and `sPos` so I make a struct. Do you think it is better to revert as before?

Reply:
done